### ISSUE TITLE
Reconsider using sympy's simplify

### ISSUE THREAD

**Issue by mstimberg** [brian-team/brian2#384]
Using sympy's `symplify` on complex expressions can take a very long time or even lead to infinite recursions. I think we should either stop using it (possibly replacing it by something more well-defined such as `factor`) or replace it by a `safe-simplify` that runs it in a separate process with a timeout, returns the unchanged expression in case of an error, etc.

**Comment by mstimberg** [comment]
I went ahead and pushed a changed that removes the `simplify` call in the explicit state updaters for now as it can make Brian go into an infinite loop (see mailing list),

**Comment by thesamovar** [comment]
I agree this is something we need to work on. The separate process idea and cacheing with joblib will help with this.

**Comment by mstimberg** [comment]
For now, we removed almost all `simplify` calls in the code. I'll leave this issue open as it would be nice to have simplification in a few places, but this needs some significant work so I'm moving the milestone.
